OpenRA.Platforms.Default/OpenGL.cs

```diff
@@ -594,16 +591,40 @@ public static void Initialize()
 				glGetTexImage = Bind<GetTexImage>("glGetTexImage");
 				glTexParameteri = Bind<TexParameteri>("glTexParameteri");
 				glTexParameterf = Bind<TexParameterf>("glTexParameterf");
-				glGenFramebuffers = Bind<GenFramebuffers>("glGenFramebuffers");
-				glBindFramebuffer = Bind<BindFramebuffer>("glBindFramebuffer");
-				glFramebufferTexture2D = Bind<FramebufferTexture2D>("glFramebufferTexture2D");
-				glDeleteFramebuffers = Bind<DeleteFramebuffers>("glDeleteFramebuffers");
-				glGenRenderbuffers = Bind<GenRenderbuffers>("glGenRenderbuffers");
-				glBindRenderbuffer = Bind<BindRenderbuffer>("glBindRenderbuffer");
-				glRenderbufferStorage = Bind<RenderbufferStorage>("glRenderbufferStorage");
-				glDeleteRenderbuffers = Bind<DeleteRenderbuffers>("glDeleteRenderbuffers");
-				glFramebufferRenderbuffer = Bind<FramebufferRenderbuffer>("glFramebufferRenderbuffer");
-				glCheckFramebufferStatus = Bind<CheckFramebufferStatus>("glCheckFramebufferStatus");
+
+				if (Profile != GLProfile.Legacy)
+				{
+					glGenVertexArrays = Bind<GenVertexArrays>("glGenVertexArrays");
+					glBindVertexArray = Bind<BindVertexArray>("glBindVertexArray");
+					glBindFragDataLocation = Bind<BindFragDataLocation>("glBindFragDataLocation");
+
+					glGenFramebuffers = Bind<GenFramebuffers>("glGenFramebuffers");
+					glBindFramebuffer = Bind<BindFramebuffer>("glBindFramebuffer");
+					glFramebufferTexture2D = Bind<FramebufferTexture2D>("glFramebufferTexture2D");
+					glDeleteFramebuffers = Bind<DeleteFramebuffers>("glDeleteFramebuffers");
+					glGenRenderbuffers = Bind<GenRenderbuffers>("glGenRenderbuffers");
+					glBindRenderbuffer = Bind<BindRenderbuffer>("glBindRenderbuffer");
+					glRenderbufferStorage = Bind<RenderbufferStorage>("glRenderbufferStorage");
+					glDeleteRenderbuffers = Bind<DeleteRenderbuffers>("glDeleteRenderbuffers");
+					glFramebufferRenderbuffer = Bind<FramebufferRenderbuffer>("glFramebufferRenderbuffer");
+					glCheckFramebufferStatus = Bind<CheckFramebufferStatus>("glCheckFramebufferStatus");
+				}
+				else
+				{
+					glGenVertexArrays = null;
+					glBindVertexArray = null;
+					glBindFragDataLocation = null;
+					glGenFramebuffers = Bind<GenFramebuffers>("glGenFramebuffersEXT");
```

Comment:
ARB_ version? it has the same semantics as core. EXT is slightly different

Reply:
EXT is what we were using for years before ripping it out, so i'm hesitant to risk changing it now!

Reply:
OK then